#ifndef TESTCOM_H
#define TESTCOM_H

#include <QDialog>
#include <QTimer>
#include <QFile>
#include <QDebug>



namespace Ui {
class TestCom;
}

class TestCom : public QDialog
{
    Q_OBJECT

public:
    explicit TestCom(QWidget *parent = nullptr);
    ~TestCom();

signals:
    void sendToCom(QByteArray data);
    void testEnd();
    void closePortTest();

public slots:
    void recieveADCdec(QString);
    void timeout_time(); // функция таймаута таймера реального времени
    void timeout_timerADC(); // функция таймаута таймера измерения ADC

private slots:
    void on_pushButton_clicked();

private:
    Ui::TestCom *ui;
    void delay(int);
    long T_ADC,T,i,x,y,l1,l2,averCounts;
    long l0 = 8388608; // - константа соответствующая разряду АЦП при нулевом напряжении;
    double ADC_V,averADC_V,ADC_offset,d;

    int warmingTime=0;
    int timeoutC2=0;
    int timeoutTest=0;
    int timeoutI=0;
    int Kintegral=0; // интеграция отсчетов Т=К*период опроса порта;
    int averNum=0;
    double R1,R2,R3,U_Ioff,I,C2,C2_1,C2_2,C2_3,C2_4;
    double DAC;
    QByteArray bytesDAC;

    QString strADC="";
    QString strDAC;

    QByteArray h00,// код реле - выкл все
               h21, // код реле - К5+К9 - замыкается 1 МОм и охранный с измерительным электроды
               h29, // код реле - К4+К5+К9 - замыкается 1 МОм + охранный с измерительным электроды + ИОН на 100 МОм(для калибровки 100 МОм)
               h15, // код реле - К2+К3+К9 - замыкается 100 МОм + охранный с измерительным электроды + ИОН на 10 ГОм(для калибровки 10 ГОм)
               h61, // код реле - К5+К6+К7+К9 - сбрасывается заряд на С1 и С2 + охранный с измерительным электроды
               h45, // код реле - К2+К6+К9 - заряжается конденсатор С2(К6) (47нФ) током от ИОН через 10 ГОм(К2) (250пА)+ охранный с измерительным электроды(К9)
               h03, // код реле - К1+К9 - замыкается 10 ГОм + охранный с измерительным электроды - измерение входного тока

               hC2, // код реле - К1+К7+К8 - Чувствительный диапазон + фильтр шумов (С1 - 560пФ)
               h82, // код реле - К1+К8 - Чувствительный диапазон
               h90, // код реле - К3+К8 - Средний диапазон (на среднем диапазоне фильтр шумов не подключается)
               hA0, // код реле - К5+К7+К8 - Грубый диапазон + фильтр шумов (С1 - 560пФ) (на грубом диапазоне фильтр шумов не отключается)
                    // + сброс заряда (по кнопке стоп) в режиме измерения заряда на чувствительном диапазоне
               h80, // код реле - К7+К8 - заряд (по кнопке старт) в режиме измерения заряда на чувствительном диапазоне
               hE0, // код реле - К5+К6+К7+К8 - сброс заряда (по кнопке стоп) в режиме измерения заряда на чувствительном диапазоне
               hC0; // код реле - К6+К8 - заряд (по кнопке старт) в режиме измерения заряда на среднем и грубом диапазоне


};

#endif // TESTCOM_H
